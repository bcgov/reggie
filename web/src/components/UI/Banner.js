//
// Reggie Web
//
// Copyright Â© 2018 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Jason Leach on 2018-09-04.
//

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Textfit } from 'react-textfit';
import logo from './bcgovlogo.svg';
import './Banner.css';

export const Banner = ({ link = '', titleText }) => {
  return (
    <div className="banner">
      <Link to={link}>
        <img src={logo} className="header-logo" alt="logo" />
      </Link>
      <Textfit className="header-title" mode="multi">
        {titleText}
      </Textfit>
    </div>
  );
};

Banner.propTypes = {
  link: PropTypes.string,
  titleText: PropTypes.string.isRequired,
};

export default Banner;
