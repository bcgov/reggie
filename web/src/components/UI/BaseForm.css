.jsform {
  display: none;
}

.toggled {
  display: block;
}

.center-form {
  display: flex;
  justify-content: center;
}

.center-form div {
  text-align: left;
}

.jsform-content {
  width: 400px;
}

.center-form #root__title {
  text-align: center;
  border-bottom: 1px solid #ced4da;
  margin-bottom: 30px;
}

@media screen and (max-width: 500px) {
  .center-form #root__title {
    font-size: 16px;
  }

  .center-form .control-label {
    font-size: 12px;
  }
}
