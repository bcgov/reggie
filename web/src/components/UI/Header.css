.header {
  background-color: #036;
  border-bottom: 2px solid #fcba19;
  color: #fff;
  padding: 0 65px;
  height: 100px;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

@media screen and (max-width: 800px) {
  .header {
    padding: 0 10px;
  }
}
