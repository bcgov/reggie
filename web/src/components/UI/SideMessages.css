.side-message {
  font-size: 13px;
  border-left: none;
  min-width: 150px;
  padding-top: 10px;
  margin-top: 40px;
  border-top: 1px solid #e5e5e5;
}

.side-message p {
  margin: 0 !important;
  color: #7c8084;
}

@media screen and (min-width: 1000px) {
  .side-message {
    margin-top: 0;
    padding-top: 0;
    border-left: 1px solid #e5e5e5;
    border-top: none;
  }
}
