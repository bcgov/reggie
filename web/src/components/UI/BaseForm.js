//
// Reggie Web
//
// Copyright Â© 2019 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Shelly Xue Han on 2019-01-30.
//

import React from 'react';
import PropTypes from 'prop-types';
import { Container, Row, Button } from 'reactstrap';
import Form from 'react-jsonschema-form';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import { Loader } from './Loader';
import './BaseForm.css';

/**
 * Json Schema Form
 *
 * @param {Object} formSchema The schema to pass into the form
 * @param {Function} formValidate Extra validation
 * @param {boolean} toggled Display the whole form or not
 * @param {Function} onSubmit The action upon clicking submit button
 * @param {Object} status The form submission status: in process (maybe other status)
 * @param {Object} messages The successful or failed msg
 * @return {Object} The form
 */
export const BaseForm = ({
  formSchema,
  formValidate = null,
  toggled,
  onSubmit,
  status,
  messages,
}) => {
  // Success or error message:
  // TODO: fix the issue as this get triggered twice, use active check for now
  if (messages.successMsg && !toast.isActive('successToast'))
    toast.success(messages.successMsg, {
      position: toast.POSITION.BOTTOM_CENTER,
      toastId: 'successToast',
      autoClose: 3000,
    });

  if (messages.failureMsg && !toast.isActive('failToast')) {
    toast.error(messages.failureMsg, {
      position: toast.POSITION.BOTTOM_CENTER,
      toastId: 'failToast',
    });
  }

  // Form:
  const jsform = (
    <Container>
      <Row className="center-form">
        <Form
          className="jsform-content"
          schema={formSchema}
          onSubmit={onSubmit}
          validate={formValidate}
          showErrorList={false}
        >
          <Button type="submit" className="btn btn-primary">
            Submit
          </Button>
        </Form>
      </Row>
    </Container>
  );

  // Alter between form and loading indication:
  const formContent = status.inProgress ? Loader : jsform;

  return (
    <div className={toggled ? 'jsform toggled' : 'jsform'}>
      <ToastContainer />
      {formContent}
    </div>
  );
};

BaseForm.propTypes = {
  formSchema: PropTypes.object.isRequired,
  toggled: PropTypes.bool.isRequired,
  onSubmit: PropTypes.func.isRequired,
  status: PropTypes.object.isRequired,
  messages: PropTypes.object.isRequired,
};
