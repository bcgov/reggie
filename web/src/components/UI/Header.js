//
// Reggie Web
//
// Copyright Â© 2018 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Jason Leach on 2018-09-04.
//

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import AuthButton from '../Auth/AuthButton';
import { APP_INFO } from '../../constants';
import Banner from './Banner';
import './Header.css';

export const Header = ({ authentication }) => {
  return (
    <header className="header">
      <Banner link="/" titleText={APP_INFO.NAME} />
      <AuthButton isAuthenticated={authentication.isAuthenticated} />
    </header>
  );
};

Header.propTypes = {
  authentication: PropTypes.object.isRequired,
};

const mapStateToProps = state => {
  return {
    authentication: state.authentication,
  };
};

export default connect(
  mapStateToProps,
  null
)(Header);
