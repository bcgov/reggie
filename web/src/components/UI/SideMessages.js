//
// Reggie Web
//
// Copyright Â© 2019 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Shelly Xue Han on 2019-01-15.
//

import React from 'react';
import PropTypes from 'prop-types';
import { Container, Row, Col } from 'reactstrap';
import './SideMessages.css';

const SideMessages = ({ centerContent, rightContent }) => {
  return (
    <Container>
      <Row>
        <Col xs="0" sm="3" />
        <Col sm="6">{centerContent}</Col>
        <Col xs="12" sm="3" className="side-message">
          {rightContent}
        </Col>
      </Row>
    </Container>
  );
};

SideMessages.propTypes = {
  leftContent: PropTypes.object,
  centerContent: PropTypes.object.isRequired,
  rightContent: PropTypes.object.isRequired,
};

SideMessages.defaultProps = {
  leftContent: null,
};

export default SideMessages;
