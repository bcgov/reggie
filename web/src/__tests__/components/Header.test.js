import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/UI/Header';

describe('Header Component', () => {
  it('matches snapshot when authenticated', () => {
    const authentication = { isAuthenticated: true };
    const wrapper = shallow(<Header authentication={authentication} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when NOT authenticated', () => {
    const authentication = { isAuthenticated: false };
    const wrapper = shallow(<Header authentication={authentication} />);
    expect(wrapper).toMatchSnapshot();
  });
});
