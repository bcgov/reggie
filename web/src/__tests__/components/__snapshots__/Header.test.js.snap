// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot when NOT authenticated 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    authentication={
      Object {
        "isAuthenticated": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Banner
          link="/"
          titleText="Pathfinder Rocket.Chat Invitation App"
        />,
        <AuthButton
          isAuthenticated={false}
        />,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "link": "/",
          "titleText": "Pathfinder Rocket.Chat Invitation App",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "isAuthenticated": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Banner
            link="/"
            titleText="Pathfinder Rocket.Chat Invitation App"
          />,
          <AuthButton
            isAuthenticated={false}
          />,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "link": "/",
            "titleText": "Pathfinder Rocket.Chat Invitation App",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isAuthenticated": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Header Component matches snapshot when authenticated 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    authentication={
      Object {
        "isAuthenticated": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Banner
          link="/"
          titleText="Pathfinder Rocket.Chat Invitation App"
        />,
        <AuthButton
          isAuthenticated={true}
        />,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "link": "/",
          "titleText": "Pathfinder Rocket.Chat Invitation App",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "isAuthenticated": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Banner
            link="/"
            titleText="Pathfinder Rocket.Chat Invitation App"
          />,
          <AuthButton
            isAuthenticated={true}
          />,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "link": "/",
            "titleText": "Pathfinder Rocket.Chat Invitation App",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isAuthenticated": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
